snuba/consumer.py

```diff
@@ -110,3 +134,209 @@ def flush_batch(self, batch: Sequence[ProcessedMessage]):
                     )
 
             self.producer.flush()
+
+
+class InsertBatchWriter(ProcessingStep[InsertBatch]):
+    def __init__(self, writer: BatchWriter[WriterTableRow]) -> None:
+        self.__writer = writer
+
+        self.__messages: MutableSequence[Message[InsertBatch]] = []
+        self.__closed = False
+
+    def poll(self) -> None:
+        pass
+
+    def submit(self, message: Message[InsertBatch]) -> None:
+        assert not self.__closed
+
+        self.__messages.append(message)
+
+    def close(self) -> None:
+        self.__closed = True
+
+        if not self.__messages:
+            return
+
+        self.__writer.write(
+            itertools.chain.from_iterable(
+                message.payload.rows for message in self.__messages
+            )
+        )
+
+    def join(self, timeout: Optional[float] = None) -> None:
+        pass
+
+
+class ReplacementBatchWriter(ProcessingStep[ReplacementBatch]):
+    def __init__(self, producer: ConfluentKafkaProducer, topic: Topic) -> None:
```

Comment:
`producer` should eventually be switched to `snuba.utils.streams.producer.Producer` to break the dependency on `ConfluentKafkaProducer` here. Last time I tried this there were some issues with ensuring the producer was properly shut down, but I haven't reinvestigated yet.